lib/logstash/plugin_mixins/elasticsearch/common.rb

```diff
@@ -154,12 +154,17 @@ def successful_connection?
       !!maximum_seen_major_version && alive_urls_count > 0
     end
 
+    def shutting_down?
+      @stopping.true? || (!execution_context.nil? && !execution_context.pipeline.nil? && execution_context.pipeline.shutdown_requested? && !execution_context.pipeline.worker_threads_draining?)
```

Comment:
To avoid the cascading `nil` checks we could use the `&.` safe navigation operator. That would simplify as:
```suggestion
      @stopping.true? || (execution_context&.pipeline&.shutdown_requested? && !execution_context&.pipeline&.worker_threads_draining?)
```